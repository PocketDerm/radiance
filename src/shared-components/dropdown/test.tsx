import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import DesktopDropdown from './desktopDropdown';
import MobileDropdown from './mobileDropdown';

import Dropdown from './index';

const options = [
  { value: 'test1', label: 'Test1' },
  { value: 'test2', label: 'Test2' },
  { value: 'test3', label: 'Test3' },
];

describe('<Dropdown />', () => {
  describe('on touch screen', () => {
    it('renders <MobileDropdown />', () => {
      window.document.documentElement.ontouchstart = () => undefined;
      const wrapper = mount(
        <Dropdown value="test1" options={options} onChange={() => null} />,
      );
      delete window.document.documentElement.ontouchstart;
      expect(wrapper.children().first().name()).toEqual('MobileDropdown');
    });
  });

  describe('when on non-touch screen', () => {
    it('renders <DesktopDropdown />', () => {
      const wrapper = mount(
        <Dropdown value="test1" options={options} onChange={() => null} />,
      );
      expect(wrapper.children().first().name()).toEqual('DesktopDropdown');
    });
  });
});

describe('<MobileDropdown />', () => {
  describe('UI snapshots', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <MobileDropdown
            onSelectChange={() => undefined}
            borderRadius="4px"
            options={options}
          />,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('onSelectChange callback', () => {
    it('should be invoked onClick', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <MobileDropdown
          borderRadius="4px"
          options={options}
          onSelectChange={spy}
          value=""
        />,
      );

      wrapper.find('select').simulate('change', { target: { value: 'test1' } });
      expect(spy).toHaveBeenCalled();
    });
  });
});

describe('<DesktopDropdown />', () => {
  it('renders the current option text', () => {
    const wrapper = mount(
      <DesktopDropdown
        borderRadius="4px"
        options={options}
        currentOption={{ value: 'test1', label: 'Test1' }}
        optionsContainerMaxHeight="250px"
        onSelectClick={() => null}
        closeDropdown={() => null}
        onOptionClick={() => null}
      />,
    );

    expect(wrapper.find('[role="button"]').text().includes('Test1')).toEqual(
      true,
    );
  });

  describe('onSelectClick callback', () => {
    it('should be invoked onClick', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <DesktopDropdown
          borderRadius="4px"
          options={options}
          currentOption={{ value: 'test1', label: 'Test1' }}
          onSelectClick={spy}
          optionsContainerMaxHeight="250px"
          closeDropdown={() => null}
          onOptionClick={() => null}
        />,
      );

      wrapper.find('[role="button"]').simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });

  describe('onOptionClick callback', () => {
    it('should be invoked onClick', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <DesktopDropdown
          borderRadius="4px"
          options={options}
          currentOption={{ value: 'test1', label: 'Test1' }}
          onOptionClick={spy}
          isOpen
          optionsContainerMaxHeight="250px"
          onSelectClick={() => null}
          closeDropdown={() => null}
        />,
      );

      wrapper.find('li').first().simulate('click');

      expect(spy).toHaveBeenCalled();
    });
  });
});
