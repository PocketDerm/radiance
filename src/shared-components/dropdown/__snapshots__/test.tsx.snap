// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileDropdown /> UI snapshots renders correctly 1`] = `
.emotion-4 {
  position: relative;
  width: 100%;
  text-align: left;
}

.emotion-4 > div:first-of-type:focus {
  outline: none;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: 0px 8px 24px rgba(52,51,82,0.05);
  background: #ffffff;
  background-image: none;
  width: 100%;
  min-height: 3.5rem;
  max-height: 3.5rem;
  border: 1px solid #ededf0;
  border-radius: 4px;
  color: #524D6E;
  line-height: 3.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  padding: 0 2rem 0 1rem;
  text-align: left;
  text-align-last: left;
  -webkit-transition: 200ms ease-in-out;
  transition: 200ms ease-in-out;
  cursor: pointer;
}

.emotion-0:hover {
  box-shadow: 0px 8px 24px rgba(52,51,82,0.10);
  -webkit-transition: 200ms ease-in-out;
  transition: 200ms ease-in-out;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 8px 24px rgba(52,51,82,0.10);
  -webkit-transition: 200ms ease-in-out;
  transition: 200ms ease-in-out;
}

.emotion-0::-moz-focus-inner {
  border: 0;
}

.emotion-2 {
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  width: 1.5rem;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.emotion-2:hover {
  cursor: pointer;
}

.emotion-2 svg {
  -webkit-transition: -webkit-transform 350ms ease-in-out;
  -webkit-transition: transform 350ms ease-in-out;
  transition: transform 350ms ease-in-out;
}

.emotion-1 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

<div
  className="emotion-4 emotion-5"
>
  <select
    className="emotion-0"
    onChange={[Function]}
    value=""
  >
    <option
      value="test1"
    >
      Test1
    </option>
    <option
      value="test2"
    >
      Test2
    </option>
    <option
      value="test3"
    >
      Test3
    </option>
  </select>
  <div
    className="emotion-2 emotion-3"
  >
    <svg
      className="emotion-1"
      fill="none"
      height={10}
      viewBox="0 0 16 16"
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.88 12.88L6 12l3.55-3.56L6 4.88 6.88 4l4.44 4.44-4.44 4.44z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
